src/modules/core/components/RadioButton/RadioButton.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useRef } from 'react';
+import { MessageDescriptor, useIntl } from 'react-intl';
+import { nanoid } from 'nanoid';
+import { useField } from 'formik';
+
+import { getMainClasses } from '~utils/css';
+import Icon from '~core/Icon';
+import { SimpleMessageValues } from '~types/index';
+
+import styles from './RadioButton.css';
+
+export interface Appearance {
+  theme?: 'primary' | 'danger';
+  direction?: 'horizontal' | 'vertical';
+}
+
+export interface RadioButtonTypes {
+  /** Appearance object */
+  appearance?: Appearance;
+  /** Disable the input */
+  disabled?: boolean;
+  /** HTML input value */
+  value: string;
+  /** Label text */
+  label: string | MessageDescriptor;
+  /** Description text values for intl interpolation */
+  labelValues?: SimpleMessageValues;
+  /** Button description */
+  description?: string | MessageDescriptor;
+  /** Description text values for intl interpolation */
+  descriptionValues?: SimpleMessageValues;
+  /** Button icon */
+  icon?: string;
+  /** If the input is checked */
+  checked: boolean;
```

Comment:
I think that `checked` prop should be excluded or made optional when you pass the `RadioButtonTypes` interface as options in `RadioButton`, otherwise it will give a type error when you instantiate it. You are not using it in `RadioButton` as prop but create a value instead.